pallets/runtime/src/asset.rs

```diff
@@ -1698,7 +1724,38 @@ impl<T: Trait> Module<T> {
         Ok(if general_status_code != ERC1400_TRANSFER_SUCCESS {
             general_status_code
         } else {
-            <percentage_tm::Module<T>>::verify_restriction(ticker, from_did, to_did, value)?
+            let mut final_result = true;
+            let mut is_valid = false;
+            let mut is_invalid = false;
+            let mut force_valid = false;
+            let tms = Self::extensions((ticker, SmartExtensionType::TransferManager))
+                .into_iter()
+                .filter(|tm| Self::extension_details((ticker, tm)).is_archive == false)
+                .collect::<Vec<T::AccountId>>();
+            if tms.len() > 0 {
+                for tm in tms.into_iter() {
+                    let result = Self::verify_restriction(
+                        ticker,
+                        extension_caller.clone(),
+                        from_did,
+                        to_did,
+                        value,
+                        tm,
+                    );
+                    match result {
+                        RestrictionResult::Valid => is_valid = true,
+                        RestrictionResult::Invalid => is_invalid = true,
+                        RestrictionResult::ForceValid => force_valid = true,
+                    }
+                }
+                //is_valid = force_valid ? true : (is_invalid ? false : is_valid);
+                final_result = force_valid || !is_invalid && is_valid;
```

Comment:
I think `final_result` should be calculated outside the loop. I don't see the need for calculating it again and again.

Reply:
It is not calculated in the loop it is always calculated outside as I am doing in the code.